File: Programmers/pro_64061_크레인 인형뽑기 게임.py
def solution(board, moves):
    #board[height][idx]
    stack = []
    answer = 0
    
    size = len(board[0])
    isFin = [0]*size
    
    for i in moves:
        if(isFin[i-1]==0):
            for j in range(size):
                if(j==size-1):
                    isFin[i-1]=1
                    
                cur = board[j][i-1]
                if(cur>0):
                    board[j][i-1]=0
                    if(stack and stack[-1]==cur):
                        answer+=2
                        stack.pop()
                    else:
                        stack.append(cur)
                    break                
    return answer

File: Programmers/pro_64061_크레인인형뽑기게임.js
function solution(board, moves) {
    const size = board.length
    const stack = []

    let answer = 0;
    moves.forEach((v, i) => {
        let col = v - 1
        for (let r = 0; r < size; r++) {
            let cur = board[r][col]
            if (cur > 0) {
                let len = stack.length
                if (len > 0 && stack[len - 1] == cur) {
                    stack.pop()
                    answer += 2
                } else
                    stack.push(board[r][col])
                board[r][col] = 0
                break
            }
        }
    })
    return answer;
}